packages/mdc-menu-surface/foundation.js

```diff
@@ -0,0 +1,572 @@
+    const availableTop = isBottomAligned ? viewportDistance.top + anchorHeight + this.anchorMargin_.bottom
+      : viewportDistance.top + this.anchorMargin_.top;
+    const availableBottom = isBottomAligned ? viewportDistance.bottom - this.anchorMargin_.bottom
+      : viewportDistance.bottom + anchorHeight - this.anchorMargin_.top;
+
+    const topOverflow = surfaceHeight - availableTop;
+    const bottomOverflow = surfaceHeight - availableBottom;
+    if (bottomOverflow > 0 && topOverflow < bottomOverflow) {
+      corner |= CornerBit.BOTTOM;
+    }
+
+    const isRtl = this.adapter_.isRtl();
+    const isFlipRtl = Boolean(this.anchorCorner_ & CornerBit.FLIP_RTL);
+    const avoidHorizontalOverlap = Boolean(this.anchorCorner_ & CornerBit.RIGHT);
+    const isAlignedRight = (avoidHorizontalOverlap && !isRtl) ||
+      (!avoidHorizontalOverlap && isFlipRtl && isRtl);
+    const availableLeft = isAlignedRight ? viewportDistance.left + anchorWidth + this.anchorMargin_.right :
+      viewportDistance.left + this.anchorMargin_.left;
+    const availableRight = isAlignedRight ? viewportDistance.right - this.anchorMargin_.right :
+      viewportDistance.right + anchorWidth - this.anchorMargin_.left;
+
+    const leftOverflow = surfaceWidth - availableLeft;
+    const rightOverflow = surfaceWidth - availableRight;
+
+    if ((leftOverflow < 0 && isAlignedRight && isRtl) ||
+        (avoidHorizontalOverlap && !isAlignedRight && leftOverflow < 0) ||
+        (rightOverflow > 0 && leftOverflow < rightOverflow)) {
+      corner |= CornerBit.RIGHT;
+    }
+
+    return corner;
+  }
+
+  /**
+   * @param {!Corner} corner Origin corner of the menu surface.
+   * @return {number} Horizontal offset of menu surface origin corner from corresponding anchor corner.
+   * @private
+   */
+  getHorizontalOriginOffset_(corner) {
+    const {anchorWidth} = this.measures_;
+    // isRightAligned corresponds to using the 'right' property on the surface.
+    const isRightAligned = Boolean(corner & CornerBit.RIGHT);
+    const avoidHorizontalOverlap = Boolean(this.anchorCorner_ & CornerBit.RIGHT);
+
+    if (isRightAligned) {
+      const rightOffset = avoidHorizontalOverlap ? anchorWidth - this.anchorMargin_.left : this.anchorMargin_.right;
+
+      // Hoisted elements positioning doesn't account for the scrollbar, so the right property needs to be reduced by
+      // the difference between the window and body width.
+      if (this.hoistedElement_ || this.isFixedPosition_) {
+        return rightOffset - (this.measures_.viewport.width - this.measures_.bodyDimensions.width);
+      }
+
+      return rightOffset;
+    }
+
+    return avoidHorizontalOverlap ? anchorWidth - this.anchorMargin_.right : this.anchorMargin_.left;
+  }
+
+  /**
+   * @param {!Corner} corner Origin corner of the menu surface.
+   * @return {number} Vertical offset of menu surface origin corner from corresponding anchor corner.
+   * @private
+   */
+  getVerticalOriginOffset_(corner) {
```

Comment:
It looks like the code here was reduced from the original code in MDC Menu, can you explain? Specifically, there used to be considerations for viewport size and distance from edges here. If I were to wager a guess, is this because we're resizing the menu instead of repositioning?

Reply:
Yes, there was extra logic here to allow the menu to go all the way to the edge of the screen and move the origin up/down based on the height of the menu. 

This behavior is not described or documented in the spec, and ignored the spec guidelines of minimum distance to the edge.